Integrations/integration-Tufin.yml

```diff
@@ -0,0 +1,321 @@
+
+        # Get Configuration
+        st_ip = demisto.params()['SecureTrack-Server']
+        st_user = demisto.params()['SecureTrack-User']['identifier']
+        st_pass = demisto.params()['SecureTrack-User']['password']
+        verify_ssl = not demisto.params().get('unsecure', False)
+        url = 'https://' + st_ip + path
+
+        # Go do
+        try:
+            res = requests.get(url, params=params, headers=headers, auth=(st_user, st_pass), verify=verify_ssl)
+        except requests.exceptions.RequestException as e:
+            return_error(str(e))
+
+        # Check output
+        if res.status_code != requests.codes.ok:
+            if res.status_code == 401:
+                return_error('TOS Reached, Auth Failed. Please check your credentials')
+            else:
+                return_error('Error {} Reaching {} to TOS: {}'.format(res.status_code, res.url, res.reason))
+        else:
+            try:
+                return res.json()
+            except:
+                return res.content
+
+
+    def path_finder():
+        """ Sample query: querystring = {"src":"10.80.80.0","dst":"172.16.200.80","service":"tcp:22","includeIncompletePaths":"true"} """
+
+        # Build the query from user input
+        querystring = {
+            "src": demisto.args()['Source'],
+            "dst": demisto.args()['Destination'],
+            "service": demisto.args().get('Service', 'Any'),
+            "includeIncompletePaths": "true"
+        }
+
+        # Define the basic output for the function, augmenting later with TOS data
+        entry = {
+            'Type': entryTypes['note'],
+            'Contents': '',
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': '',
+            'EntryContext': {}
+        }
+
+        # Ask TOS for the path
+        o = tos_request("/securetrack/api/topology/path", querystring)
+
+        # Verify the data and return
+        try:
+            entry['EntryContext']['Tufin.TrafficAllowed'] = o['path_calc_results']['traffic_allowed']
+            entry['EntryContext']['Tufin.TrafficRawJSON'] = o['path_calc_results']['device_info']
+            entry['EntryContext']['Tufin.TrafficDevices'] = [d['name'] for d in o['path_calc_results']['device_info']]
+            entry['Contents'] = o['path_calc_results']['device_info']
+            entry['HumanReadable'] = tableToMarkdown('Tufin Topology Search for {} to {} via Service {}. Traffic is {}'.format(querystring['src'], querystring['dst'], querystring['service'], ('**Denied**', '**Allowed**')[o['path_calc_results']['traffic_allowed']]),
+            {'Start': querystring['src'], 'Devices in Path' : "-->".join(['**' + d['name'] + '**' + ' ({})'.format(d['vendor']) for d in o['path_calc_results']['device_info']]), 'End': querystring['dst']}, ['Start', 'Devices in Path', 'End'])
+        except KeyError:
+            entry['HumanReadable'] = 'Unknown Output Returned'
```

Comment:
Please use `return_error()` function instead

Reply:
What is the purpose of entryTypes?  e.g. entry['Type'] = entryTypes['error'] 
vs.
return_error

Will modify in next commit